tests/testsuite/shell_completions/mod.rs

```diff
@@ -1,2 +1,6 @@
 pub mod common;
 
+mod bash;
+mod elvish;
+mod fish;
+mod zsh;
```

Comment:
Do we need these to be separate files?  We do not need to exhaustively test completions.  I'm not even sure we need to test every shell.  Our primary aim is to test that the `CompleteEnv` call is in place and works as expected.

For shell-particulars or native completion particulars, `clap_complete`s tests are covering us.  For our custom completers, we should probably skip end-to-end tests for those and do integration tests.